### DISCUSSION TITLE
tdslite with ENC28J60 ?

### DISCUSSION THREAD

**Discussion by kangyite** [tdslite/tdslite#27] [Q&A]
Is it possible to use ENC28J60 instead of the ethernet shield?  The library for ENC28J60 is [https://github.com/njh/EtherCard](url).

Thanks

**Comment by xmkg** [comment]
AFAIK, the EtherCard library is not a drop-in replacement for the standard Arduino Ethernet library, it does not implement the Arduino's EthernetClient interface, so it cannot work with tdslite out of the box. I believe you can try using [UIPEthernet](https://github.com/UIPEthernet/UIPEthernet) instead of EtherCard, which is another library for ENC28J60 and it supports the said interface. Let me know how it goes.
